news/views.py

```diff
@@ -46,6 +49,42 @@
     newsletter_exception_response)
 
 
+def ip_rate_limit_key(group, request):
+    return request.META.get('HTTP_X_CLUSTER_CLIENT_IP',
+                            request.META.get('REMOTE_ADDR'))
+
+
+def ip_rate_limit_rate(group, request):
+    client_ip = ip_rate_limit_key(group, request)
+    if client_ip and client_ip.startswith('10.'):
+        # internal request, no limit.
+        return None
```

Comment:
I'm wondering if it might make more sense to set a high limit for internal requests instead of none at all.